src/main/java/domain/AttendanceRecord.java

```diff
@@ -0,0 +1,32 @@
+package domain;
+
+import static domain.AttendanceResult.ABSENCE;
+import static domain.AttendanceResult.ATTENDANCE;
+import static domain.AttendanceResult.LATE;
+
+import java.util.EnumMap;
+import java.util.Map;
+
+public class AttendanceRecord {
+    private final int attendanceCount;
+    private final int lateCount;
+    private final int absenceCount;
+
+    public AttendanceRecord(Map<AttendanceResult, Integer> record) {
+        this.attendanceCount = record.getOrDefault(ATTENDANCE, 0);
+        this.lateCount = record.getOrDefault(LATE, 0);
+        this.absenceCount = record.getOrDefault(ABSENCE, 0);
+    }
+
+    public AbsencePenalty calculateAbsencePenalty() {
+        return AbsencePenalty.findAbsenceLevel(absenceCount, lateCount);
+    }
+
+    public Map<AttendanceResult, Integer> getTotalAttendanceRecord() {
+        Map<AttendanceResult, Integer> result = new EnumMap<>(AttendanceResult.class);
+        result.put(ATTENDANCE, attendanceCount);
+        result.put(LATE, lateCount);
+        result.put(ABSENCE, absenceCount);
+        return result;
```

Comment:
아래와 같이 [컬렉션 팩토리](https://devbksheen.tistory.com/entry/%EB%AA%A8%EB%8D%98-%EC%9E%90%EB%B0%94-%EC%BB%AC%EB%A0%89%EC%85%98-%ED%8C%A9%ED%86%A0%EB%A6%ACCollection-Factory-%EC%86%8C%EA%B0%9C-%EB%B0%8F-%EC%82%AC%EC%9A%A9%EB%B2%95)를 활용해볼 수 있을 것 같아요. 

```suggestion
        return Map.of(
            ATTENDANCE, attendanceCount,
            LATE, lateCount,
            ABSENCE, absenceCount
        );
```

Reply:
확실히 깔끔하네요! 개선된 방법 알려주셔서 감사합니다 🥹